lib/src/common/repos_service.dart

```diff
@@ -469,71 +684,236 @@ class RepositoriesService extends Service {
-  Future<Release> getRelease(RepositorySlug slug, int id) =>
-      _github.getJSON("/repos/${slug.fullName}/releases/$id",
-          convert: Release.fromJson);
+  Future<Release> getRelease(RepositorySlug slug, int id) async =>
+      _github.getJSON<Map<String, dynamic>, Release>(
+        "/repos/${slug.fullName}/releases/$id",
+        convert: (i) => Release.fromJson(i),
+      );
 
   /// Creates a Release based on the specified [release].
   ///
   /// API docs: https://developer.github.com/v3/repos/releases/#create-a-release
-  Future<Release> createRelease(RepositorySlug slug, CreateRelease release) {
-    return _github.postJSON("/repos/${slug.fullName}/releases",
-        convert: Release.fromJson, body: jsonEncode(release.toJson()));
+  Future<Release> createRelease(
+      RepositorySlug slug, CreateRelease release) async {
+    return _github.postJSON<Map<String, dynamic>, Release>(
+      "/repos/${slug.fullName}/releases",
+      convert: (i) => Release.fromJson(i),
+      body: jsonEncode(release.toJson()),
+    );
+  }
+
+  /// Edits the given release with new fields.
+  /// * [tagName]: The name of the tag.
+  /// * [targetCommitish]: Specifies the commitish value that determines where the Git tag is created from. Can be any branch or commit SHA. Unused if the Git tag already exists. Default: the repository's default branch (usually master).
+  /// * [name]: The name of the release.
+  /// * [body]: Text describing the contents of the tag.
+  /// * [draft]: true makes the release a draft, and false publishes the release.
+  /// * [preRelease]: true to identify the release as a prerelease, false to identify the release as a full release.
+  ///
+  /// Leave blank the fields you don't want to edit.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#edit-a-release
+  Future<Release> editRelease(
+    RepositorySlug slug,
+    Release releaseToEdit, {
+    String tagName,
+    String targetCommitish,
+    String name,
+    String body,
+    bool draft,
+    bool preRelease,
+  }) async {
+    return _github.postJSON<Map<String, dynamic>, Release>(
+      "/repos/${slug.fullName}/releases/${releaseToEdit.id.toString()}",
+      body: jsonEncode(createNonNullMap(<String, dynamic>{
+        "tag_name": tagName ?? releaseToEdit.tagName,
+        "target_commitish": targetCommitish ?? releaseToEdit.targetCommitish,
+        "name": name ?? releaseToEdit.name,
+        "body": body ?? releaseToEdit.body,
+        "draft": draft ?? releaseToEdit.draft,
+        "prerelease": preRelease ?? releaseToEdit.prerelease,
+      })),
+      statusCode: StatusCodes.OK,
+      convert: (i) => Release.fromJson(i),
+    );
+  }
+
+  /// Delete the release.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#delete-a-release
+  Future<void> deleteRelease(RepositorySlug slug, Release release) async {
+    await _github.request(
+      "DELETE",
+      "/repos/${slug.fullName}/releases/${release.id.toString}",
```

Comment:
I think this may be a bug. You aren't calling toString, you're referencing the method. It should probably be:
```suggestion
      "/repos/${slug.fullName}/releases/${release.id}",
```